crypto/err/err_prn.c

```diff
@@ -12,24 +12,30 @@
 #include <openssl/crypto.h>
 #include <openssl/buffer.h>
 #include <openssl/err.h>
+#include "err_locl.h"
 
 void ERR_print_errors_cb(int (*cb) (const char *str, size_t len, void *u),
                          void *u)
 {
     CRYPTO_THREAD_ID tid = CRYPTO_THREAD_get_current_id();
     unsigned long l;
-    char buf[256];
-    char buf2[4096], *hex;
-    const char *file, *data;
+    char buf[4096], *hex;
+    const char *lib, *reason;
+    const char *file, *data, *func;
     int line, flags;
 
-    while ((l = ERR_get_error_line_data(&file, &line, &data, &flags)) != 0) {
-        ERR_error_string_n(l, buf, sizeof(buf));
+    while ((l = ERR_get_error_all(&file, &line, &func, &data, &flags)) != 0) {
+        lib = ERR_lib_error_string(l);
+        reason = ERR_reason_error_string(l);
+        if (func == NULL)
+            func = "unknown function";
```

Comment:
is "internal" or "private" a better term?  we could know what it is, we're just not saying :)

Reply:
Well, there are two possibilities here:

1. `no-filenames` was configured, and in that case, I would frankly prefer it to be nothing, i.e. the empty string (which means changing the definition of `OPENSSL_FUNC` when `OPENSSL_NO_FILENAMES` is defined, thanks for pointing that out)
2. There is no function name to be had, for unknown reasons, i.e. `ERR_set_debug` was called with NULL for the function name.  In that case, I think "unknown function" is as appropriate than whatever else...